docker-compose.yml

```diff
@@ -1745,6 +1746,34 @@ services:
       /bin/bash -c "
         /arrow/ci/scripts/r_sanitize.sh /arrow"
 
+  r-clang-ubsan:
+    image: ${REPO}:r-rhub-clang-ubsan-devel-latest
+    build:
+      context: .
+      dockerfile: ci/docker/linux-r.dockerfile
+      cache_from:
+        - ${REPO}:r-rhub-clang-ubsan-devel-latest
+      args:
+        base: rhub/clang-ubsan
+        cmake: ${CMAKE}
+        r_dev: ${ARROW_R_DEV}
+        r_bin: R
+        tz: ${TZ}
+        r_prune_deps: ${R_PRUNE_DEPS}
+    shm_size: *shm-size
+    environment:
+      <<: *common
+      LIBARROW_DOWNLOAD: "false"
+      ARROW_SOURCE_HOME: "/arrow"
+      ARROW_R_DEV: ${ARROW_R_DEV}
+      # To test for CRAN release, delete ^^ these two env vars so we download the Apache release
```

Comment:
Hrm, I'm worried about forgetting to do this step. Is there maybe a more foolproof way to do what we want here?

Reply:
Works for me, thanks!